src/hotspot/share/cds/cdsProtectionDomain.cpp

```diff
@@ -199,24 +199,27 @@ Handle CDSProtectionDomain::get_shared_jar_manifest(int shared_path_index, TRAPS
 Handle CDSProtectionDomain::get_shared_jar_url(int shared_path_index, TRAPS) {
   Handle url_h;
   if (shared_jar_url(shared_path_index) == nullptr) {
-    JavaValue result(T_OBJECT);
     const char* path = FileMapInfo::shared_path_name(shared_path_index);
-    Handle path_string = java_lang_String::create_from_str(path, CHECK_(url_h));
-    Klass* classLoaders_klass =
-        vmClasses::jdk_internal_loader_ClassLoaders_klass();
-    JavaCalls::call_static(&result, classLoaders_klass,
-                           vmSymbols::toFileURL_name(),
-                           vmSymbols::toFileURL_signature(),
-                           path_string, CHECK_(url_h));
-
-    atomic_set_shared_jar_url(shared_path_index, result.get_oop());
+    oop result_oop = get_shared_jar_url_helper(path, url_h, CHECK_(url_h));
+    atomic_set_shared_jar_url(shared_path_index, result_oop);
   }
 
   url_h = Handle(THREAD, shared_jar_url(shared_path_index));
   assert(url_h.not_null(), "sanity");
   return url_h;
 }
 
+oop CDSProtectionDomain::get_shared_jar_url_helper(const char* path, Handle url_h, TRAPS) {
```

Comment:
```suggestion
oop CDSProtectionDomain::get_file_URL(const char* path, Handle url_h, TRAPS) {
```
It looks odd to have a "helper" method as part of the public API, especially when called directly.

Reply:
I think both function can use the same name, since they have overloaded parameter types:

```
Handle CDSProtectionDomain::get_shared_jar_url(int shared_path_index, TRAPS);
oop CDSProtectionDomain::get_shared_jar_url(const char* path, Handle url_h, TRAPS);

```

Reply:
I think using the same name might be confusing here since they return totally different things. I prefer David's suggestion here since all the method is doing is calling `toFileURL`.